src/Elastic.Clients.Elasticsearch/Serialization/UnionConverter.cs

```diff
@@ -122,12 +122,6 @@ private class DerivedUnionConverterInner<TType, TItem1, TItem2> : JsonConverter<
 		public override void Write(Utf8JsonWriter writer, TType value,
 			JsonSerializerOptions options)
 		{
-			if (value is null)
```

Comment:
Looks like there's a performance optimization in collection converters which can cause an issue if it contains  `null` value, see dotnet/runtime#85981. Will bring it back.

Reply:
Thought that there's a breaking change, so verified how `null`s are handled. So, if `HandleNull` isn't specified then no need to worry about `value` being `null` ([source](https://github.com/dotnet/runtime/blob/c5256458235e00c4bc4634c3cc389fefdd4c10d8/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs#L373-L379):

```csharp
            if (default(T) is null && !HandleNullOnWrite && IsNull(value))
            {
                // We do not pass null values to converters unless HandleNullOnWrite is true. Null values for properties were
                // already handled in GetMemberAndWriteJson() so we don't need to check for IgnoreNullValues here.
                writer.WriteNullValue();
                return true;
            }
```